internal/packages/packages.go

```diff
@@ -243,10 +244,17 @@ func (dsm *DataStreamManifest) GetPipelineNameOrDefault() string {
 // for this data stream.
 func (dsm *DataStreamManifest) IndexTemplateName(pkgName string) string {
 	if dsm.Dataset == "" {
-		return fmt.Sprintf("%s-%s.%s", dsm.Type, pkgName, dsm.Name)
+		return dsm.IndexTemplateNamePrefix() + fmt.Sprintf("%s-%s.%s", dsm.Type, pkgName, dsm.Name)
```

Comment:
What would be the value for a hidden data stream without dataset defined? `.logs-aPackage.aDataStream`?

Reply:
Not 100% sure what it suppose to be, but currently the kibana behavour is different from the elatic-package validation. Kibana creates templates with "." dot prefix for the package with hidden datastreams.

Reply:
> Not 100% sure what it suppose to be,

So do I :) I'm not if it's a bug or works as intended. @ruflin @joshdover Could you please confirm the index template name ("hidden" case)?

Reply:
So this pre-dates my time on Fleet, but this definitely appears to be intentional. We have this function in Kibana which is used for all asset naming:

https://github.com/elastic/kibana/blob/0ee514b1fe044cdaca253f8f14988fc17635fed3/x-pack/plugins/fleet/server/services/epm/elasticsearch/index.ts#L10-L18

It appears it was added as part of https://github.com/elastic/kibana/pull/86277 over a year ago, though I don't see any linked package-spec changes. Maybe @kevinlog can provide more context.

Reply:
Thanks for diving into code, Josh! @aleksmaus I think you can adjust function implementation to look similar (`baseType` vs `.baseType`) and please hard-link to Kibana in the comment.

Reply:
Thanks for looking into this. Will update and add comment. Endpoint didn't have issues passing this validation in CI because it lives in a separate repo, maybe skipping this test.

Reply:
Took a look at kibana code, it looks consistent with this change. Added a comment. Let me know.